### PR TITLE
blob/blobs - porting over to hashicorp/go-azure-sdk's base layer

### PR DESCRIPTION
**Pull request by catriona-m**

### CODE REVIEW

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @catriona-m 

Thanks for this PR - apologies for the delayed review here.

I've taken a look through and left some comments inline, but on the whole this is looking pretty good - if we can fix those up this should otherwise be good to go 👍 

Thanks!

File: storage/2020-08-04/blob/blobs/api.go [hunk 1]
```diff
@@ -4,42 +4,39 @@ import (
...
-	GetResourceID(accountName, containerName, blobName string) string
-	SetTier(ctx context.Context, accountName, containerName, blobName string, tier AccessTier) (result autorest.Response, err error)
-	Snapshot(ctx context.Context, accountName, containerName, blobName string, input SnapshotInput) (result SnapshotResult, err error)
-	GetSnapshotProperties(ctx context.Context, accountName, containerName, blobName string, input GetSnapshotPropertiesInput) (result GetPropertiesResult, err error)
-	Undelete(ctx context.Context, accountName, containerName, blobName string) (result autorest.Response, err error)
+	AppendBlock(ctx context.Context, containerName string, blobName string, input AppendBlockInput) (AppendBlockResponse, error)
+	Copy(ctx context.Context, containerName string, blobName string, input CopyInput) (CopyResponse, error)
+	AbortCopy(ctx context.Context, containerName string, blobName string, input AbortCopyInput) (CopyAbortResponse, error)
+	CopyAndWait(ctx context.Context, containerName string, blobName string, input CopyInput, pollingInterval time.Duration) error
```
[the remark this answers is not in the document]
Comment (catriona-m) [reply storage/2020-08-04/blob/blobs/api.go:13]: will it be ok to add this in a future pr with the file poller?

[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply storage/2020-08-04/blob/blobs/api.go:13]: Yep, sounds good 👍

File: storage/2020-08-04/blob/blobs/properties_get.go [hunk 2]
```diff
@@ -163,148 +161,127 @@ type GetPropertiesResult struct {
...
 			}
 
-			result.ContentLength = int64(i)
+			if v := resp.HttpResponse.Header.Get("x-ms-server-encrypted"); v != "" {
+				b, innerErr := strconv.ParseBool(v)
+				if innerErr != nil {
+					err = fmt.Errorf("error parsing %q as a bool: %s", v, innerErr)
+					return
+				}
+				resp.IncrementalCopy = b
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply storage/2020-08-04/blob/blobs/properties_get.go:260]: Looks like this one is still pending?

[the remark this answers is not in the document]
Comment (catriona-m) [reply storage/2020-08-04/blob/blobs/properties_get.go]: oops!! fixed now!

**Review by tombuildsstuff — APPROVED** [review]

**Pull request merged**